perception/autoware_tensorrt_bevdet/include/autoware/tensorrt_bevdet/bevdet_node.hpp

```diff
@@ -0,0 +1,221 @@
+void getTransform(
+  const geometry_msgs::msg::TransformStamped & transform, Eigen::Quaternion<float> & rot,
+  Eigen::Translation3f & translation);
+
+// Get the camera intrinsics from a sensor_msgs::msg::CameraInfo
+void getCameraIntrinsics(
+  const sensor_msgs::msg::CameraInfo::SharedPtr msg, Eigen::Matrix3f & intrinsics);
+
+class TRTBEVDetNode : public rclcpp::Node
+{
+  /**
+   * @class TRTBEVDetNode
+   * @brief This class represents a node for performing object detection using TensorRT on BEV
+   * (Bird's Eye View) images.
+   */
+private:
+  size_t img_N_;    ///< Number of images
+  uint32_t img_w_;  ///< Width of the images
+  uint32_t img_h_;  ///< Height of the images
+
+  std::string model_config_;  ///< Path to the model configuration file
+
+  std::string onnx_file_;    ///< Path to the ONNX file
+  std::string engine_file_;  ///< Path to the TensorRT engine file
+
+  std::vector<std::string> imgs_name_;    ///< Names of the images
+  std::vector<std::string> class_names_;  ///< Names of the object classes
+
+  camsData sampleData_;  ///< Sample data for camera parameters
+  std::shared_ptr<autoware::tensorrt_bevdet::BEVDet>
+    bevdet_;  ///< Object for performing object detection
+
+  uchar * imgs_dev_ = nullptr;  ///< Device pointer for storing the images
+  float score_thre_;            ///< Score threshold for object detection
+
+  rclcpp::Publisher<autoware_perception_msgs::msg::DetectedObjects>::SharedPtr
+    pub_boxes_;  ///< Publisher for publishing the detected objects
+
+  // Subscribers of camera info for each camera, no need to synchronize
+  rclcpp::Subscription<sensor_msgs::msg::CameraInfo>::SharedPtr
+    sub_f_caminfo_;  ///< Subscriber for front camera info
+  rclcpp::Subscription<sensor_msgs::msg::CameraInfo>::SharedPtr
+    sub_fl_caminfo_;  ///< Subscriber for front-left camera info
+  rclcpp::Subscription<sensor_msgs::msg::CameraInfo>::SharedPtr
+    sub_fr_caminfo_;  ///< Subscriber for front-right camera info
+  rclcpp::Subscription<sensor_msgs::msg::CameraInfo>::SharedPtr
+    sub_b_caminfo_;  ///< Subscriber for back camera info
+  rclcpp::Subscription<sensor_msgs::msg::CameraInfo>::SharedPtr
+    sub_bl_caminfo_;  ///< Subscriber for back-left camera info
+  rclcpp::Subscription<sensor_msgs::msg::CameraInfo>::SharedPtr
+    sub_br_caminfo_;  ///< Subscriber for back-right camera info
+  std::vector<bool>
+    caminfo_received_;  ///< Flag indicating if camera info has been received for each camera
+  bool camera_info_received_flag_ =
+    false;                    ///< Flag indicating if camera info has been received for all cameras
+  bool initialized_ = false;  /// Flag indicating if img_w_ and img_h_ has been initialized
+
+  // tf listener
+  std::shared_ptr<tf2_ros::TransformListener> tf_listener_{
+    nullptr};                                   ///< TF listener for transforming coordinates
+  std::unique_ptr<tf2_ros::Buffer> tf_buffer_;  ///< Buffer for storing TF transforms
```

Comment:
The `caminfo_received_` vector is used to track camera info reception.  However, its size is not explicitly set or tied to the number of cameras.  It's crucial to ensure this vector is properly sized to avoid out-of-bounds errors.  Consider initializing its size in the constructor based on the number of cameras the node is configured to use. For example: `caminfo_received_.resize(num_cameras);` where `num_cameras` is a parameter determined in the constructor.